datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedPartitionAssigner.java

```diff
@@ -5,30 +5,128 @@
  */
 package com.linkedin.datastream.server.assignment;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.stream.Collectors;
 
-import org.apache.commons.lang.NotImplementedException;
+import com.google.common.annotations.VisibleForTesting;
 
 import com.linkedin.datastream.server.ClusterThroughputInfo;
+import com.linkedin.datastream.server.DatastreamGroupPartitionsMetadata;
 import com.linkedin.datastream.server.DatastreamTask;
+import com.linkedin.datastream.server.DatastreamTaskImpl;
+import com.linkedin.datastream.server.PartitionThroughputInfo;
 
 
 /**
  * Performs partition assignment based on partition throughput information
```

Comment:
Do we really need a separate estimator and assigner class?

Reply:
- The reason for a separate assigner is to be able to implement and plug in different assignment algorithms if needed.
- The reason for having a separate task count estimator is that at some point we might want to refactor different estimation mechanisms (throughput based, elastic) and make them all implement the same interface.